packages/app/src/useConfirm.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { atom, useAtomValue, useSetAtom } from 'jotai';
+import Modal from 'react-bootstrap/Modal';
+import Button from 'react-bootstrap/Button';
+
+type ConfirmAtom = {
+  message: string;
+  confirmLabel?: string;
+  onConfirm: () => void;
+  onClose?: () => void;
+} | null;
+
+const confirmAtom = atom<ConfirmAtom>(null);
+
+export const useConfirm = () => {
+  const setConfirm = useSetAtom(confirmAtom);
+
+  return React.useCallback(
+    async (message: string, confirmLabel?: string): Promise<boolean> => {
+      return new Promise(resolve => {
+        setConfirm({
+          message,
+          confirmLabel,
+          onConfirm: () => {
+            resolve(true);
+            setConfirm(null);
+          },
+          onClose: () => {
+            resolve(false);
+            setConfirm(null);
+          },
+        });
+      });
+    },
+    [setConfirm],
+  );
+};
+
+export const useConfirmModal = () => {
+  const confirm = useAtomValue(confirmAtom);
+  const setConfirm = useSetAtom(confirmAtom);
+
+  const handleClose = React.useCallback(() => {
+    confirm?.onClose?.();
+    setConfirm(null);
+  }, [confirm, setConfirm]);
+
+  return confirm ? (
+    <Modal show onHide={handleClose}>
```

Comment:
Personally I'd prefer a tooltip next to the action button but can see how this is faster to implement rn